src/particles.js

```diff
@@ -74,8 +74,64 @@ class TVar {
 class Particle {
   constructor (net) {
     this.net = net
-    this.x = Math.random() * net.limx
-    this.y = Math.random() * net.limy
+    this.val = Math.random()
+
+    if (net.opts.edgeOnly)
+    {
+      this.threshold1 = 0.5
+      this.threshold2 = 1.0 
+
+      if (net.opts.edgeAndCentre)
+      {
+        this.threshold1 = 0.3
+        this.threshold2 = 0.6 
+      }
+
+      if (Math.random() <= 0.5)
```

Comment:
Maybe just throw in a few helpful comments here like

"top particle", "left particle", "center vertical"... etc.